### DISCUSSION TITLE
Order of list of objects

### DISCUSSION THREAD

**Discussion by atasoglou** [nlohmann/json#3273] [Q&A] [answered]
Hello.
I can see [what it is assumed for the name/value  order](https://json.nlohmann.me/features/object_order/), however I couldn't find somewhere what the order of a list of json objects would be. For example:

```
#include <iostream>
#include "json.hpp"

using json = nlohmann::json;

int main()
{
    json j1;
    j["a"] = 1;
    j["b"] = 2;
    j["c"] = 3;

    json j2;
    j["a"] = 30;
    j["b"] = 10;
    j["c"] = 50;

   // etc jn

   json mainJson;
   mainJson["date"] = "2022-01-14";
   mainJson["list"].push_back(j1);
   mainJson["list"].push_back(j2);
   ...
   mainJson["list"].push_back(jn);
   
   // How is the order of the "list" defined?
   std::cout << j.dump(mainJson) << '\n';
}
```
Is the order of inserting objects `jn` kept or is this also being re-ordered somehow?

Thanks!

**Comment by nlohmann** [comment, marked as answer]
The order of an array is maintained no matter of the type of the stored values.
